staging/vhost-device-gpu/src/virt_gpu.rs

```diff
@@ -583,11 +625,46 @@ impl VirtioGpu for RutabagaVirtioGpu {
         Ok(OkNoData)
     }
 
-    fn flush_resource(&mut self, resource_id: u32) -> VirtioGpuResult {
+    /// If the resource is the scanout resource, flush it to the display.
+    fn flush_resource(
+        &mut self,
+        resource_id: u32,
+        gpu_backend: &mut GpuBackend,
+        rect: Rectangle,
+    ) -> VirtioGpuResult {
         if resource_id == 0 {
             return Ok(OkNoData);
         }
 
+        let resource = self
+            .resources
+            .get(&resource_id)
+            .ok_or(ErrInvalidResourceId)?;
+
+        for scanout_id in resource.scanouts.iter_enabled() {
+            let mut image = vec![0; rect.width as usize * rect.height as usize * 4];
```

Comment:
why not use a constant defining the number of bytes per pixel instead of `4` so that it is clear, also the name image seems misleading how about `RGB data` or `data` wdyt?

Reply:
Yes right, I added a constant. 
I renamed the variable to `data`, `rgb_data` a slightly confusing to me because the name seems to suggest it would be 3 bytes per pixel. The formats qemu is expecting for scanout is XRGB (PIXMAN_x8r8g8b8 according to the spec).